### PR TITLE
DELETE FileShare API Support

### LINKED ISSUE #993: Delete a Fileshare in AWS
**Issue/Feature Description:**
API to delete the fileshare in AWS
[DELETE] v1/\<tenantId\>/Backend/\<backendId\>/shares/\<shareId\>
**Why this issue to fixed / feature is needed(give scenarios or use cases):**
User may want to delete the Fileshare which is not required because of different reasons
**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by himanshuvar**
**What type of PR is this?**

> /kind new feature

**What this PR does / why we need it**:
DELETE API support for FileShare to delete the fileshare
`[DELETE] /v1/<tenantId>/file/shares/<id>`

**Which issue(s) this PR fixes**:

Fixes #993 #997 

**Test Report Added?**:

> /kind TESTED

**Test Report**:

The following testing is done for DELETE File Share API:

- [x] List Backend

- [x] Get FileShare

- [x] List File Share for Azure Backend

- [x] List File Share for AWS Backend

- [x] Delete FileShare

- [x] List File Share for AWS Backend After Delete

- [x] List all File Share

```
List Backends:

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/backends

{
    "backends": [
        {
            "id": "5eecefa90de5c3000151d1d4",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "name": "azure-backend-file",
            "type": "azure-file",
            "region": "East Asia",
            "endpoint": "https://myaccount.file.core.windows.net/",
            "bucketName": "container"
        },
        {
            "id": "5eecefe30de5c3000151d1d5",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "name": "aws-backend-file",
            "type": "aws-file",
            "region": "ap-south-1",
            "endpoint": "s3.amazonaws.com",
            "bucketName": "opensds-him"
        }
    ],
    "next": 2
}

Get File Share

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares/5eecf099a4fb8b25102593a4

{
            "id": "5eecf099a4fb8b25102593a4",
            "createdAt": "2020-06-11T01:12:32.662883Z",
            "updatedAt": "2020-06-11T01:12:34.682329Z",
            "name": "azure-fileshareDemo",
            "description": "Azure FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefa90de5c3000151d1d4",
            "backend": "azure-backend-file",
            "type": "generalPurpose",
            "region": "Southeast Asia",
            "availabilityZone": "Southeast Asia",
            "status": "success"
  }

List File Share for Azure Backend:

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares?backendId=5eecefa90de5c3000151d1d4

{
    "fileshares": [
        {
            "id": "5eecf099a4fb8b25102593a4",
            "createdAt": "2020-06-11T01:12:32.662883Z",
            "updatedAt": "2020-06-11T01:12:34.682329Z",
            "name": "azure-fileshareDemo",
            "description": "Azure FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefa90de5c3000151d1d4",
            "backend": "azure-backend-file",
            "type": "generalPurpose",
            "region": "Southeast Asia",
            "availabilityZone": "Southeast Asia",
            "status": "success"
        }
    ],
    "next": 1
}

List File Share for AWS Backend:

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares?backendId=5eecefe30de5c3000151d1d5

{
    "fileshares": [
        {
            "id": "5eecf38aa4fb8b25102593a5",
            "createdAt": "2020-06-11T01:12:29.662883Z",
            "updatedAt": "2020-06-11T01:12:31.682329Z",
            "name": "aws-fileshareDemo",
            "description": "AWS FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefe30de5c3000151d1d5",
            "backend": "aws-backend-file",
            "type": "bursting",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1a",
            "status": "success",
            "tags": [
                {
                    "key": "Name",
                    "value": "FSName"
                },
                {
                    "key": "Department",
                    "value": "Billing"
                }
            ]
        },
        {
            "id": "5eecf3c0a4fb8b25102593a6",
            "createdAt": "2020-06-11T01:12:29.662883Z",
            "updatedAt": "2020-06-11T01:12:31.682329Z",
            "name": "aws-fileshare-test",
            "description": "AWS FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefe30de5c3000151d1d5",
            "backend": "aws-backend-file",
            "type": "generalPurpose",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1b",
            "status": "success",
            "tags": [
                {
                    "key": "Name",
                    "value": "TestFS"
                },
                {
                    "key": "Department",
                    "value": "Testing"
                }
            ]
        }
    ],
    "next": 2
}

Delete File Share

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares/5eecf3c0a4fb8b25102593a6

200 OK
{ } 

List File Share for AWS Backend After Delete:

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares?backendId=5eecefe30de5c3000151d1d5

{
    "fileshares": [
        {
            "id": "5eecf38aa4fb8b25102593a5",
            "createdAt": "2020-06-11T01:12:29.662883Z",
            "updatedAt": "2020-06-11T01:12:31.682329Z",
            "name": "aws-fileshareDemo",
            "description": "AWS FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefe30de5c3000151d1d5",
            "backend": "aws-backend-file",
            "type": "bursting",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1a",
            "status": "success",
            "tags": [
                {
                    "key": "Name",
                    "value": "FSName"
                },
                {
                    "key": "Department",
                    "value": "Billing"
                }
            ]
        }
    ],
    "next": 1
}

List all File Share:

http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares

{
    "fileshares": [
        {
            "id": "5eecf099a4fb8b25102593a4",
            "createdAt": "2020-06-11T01:12:32.662883Z",
            "updatedAt": "2020-06-11T01:12:34.682329Z",
            "name": "azure-fileshareDemo",
            "description": "Azure FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefa90de5c3000151d1d4",
            "backend": "azure-backend-file",
            "type": "generalPurpose",
            "region": "Southeast Asia",
            "availabilityZone": "Southeast Asia",
            "status": "success"
        },
        {
            "id": "5eecf38aa4fb8b25102593a5",
            "createdAt": "2020-06-11T01:12:29.662883Z",
            "updatedAt": "2020-06-11T01:12:31.682329Z",
            "name": "aws-fileshareDemo",
            "description": "AWS FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5eecefe30de5c3000151d1d5",
            "backend": "aws-backend-file",
            "type": "bursting",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1a",
            "status": "success",
            "tags": [
                {
                    "key": "Name",
                    "value": "FSName"
                },
                {
                    "key": "Department",
                    "value": "Billing"
                }
            ]
        }
    ],
    "next": 2
}



<!--
*Please provide the test report link (public accessible, screen shot or copy paste the test report, 
or add the testing details.
-->

**Special notes for your reviewer**:

### CODE REVIEW

File: file/pkg/service/service.go [hunk 1]
```diff
@@ -80,3 +82,55 @@ func (f *fileService) ListFileShare(ctx context.Context, in *pb.ListFileShareReq
...
+		EncryptionSettings: fs.EncryptionSettings,
+		Metadata:           fs.Metadata,
+	}
+	log.Info("Get file share successfully.")
+	return nil
+}
+
+func (f *fileService) DeleteFileShare(ctx  context.Context, in *pb.DeleteFileShareRequest, out *pb.DeleteFileShareResponse) error {
+	log.Info("Received DeleteFileShare request.")
+	err := db.DbAdapter.DeleteFileShare(ctx, in.Id)
```
Comment (sfzeng) [inline file/pkg/service/service.go:129]: Seems only delete the recorde from database, not delete it from the cloud backend, need to take care of it.

Comment (himanshuvar) [reply file/pkg/service/service.go:129]: @sfzeng According to current plan. Delete API should delete the FS object from multi-cloud FS db. 
The FS created can be in usage through different mount targets. So deleting FS at cloud backend is not a good idea before make these checks like the usage of FS.

Comment (sfzeng) [reply file/pkg/service/service.go:129]: ok, but if we delete it without considering the cloud backend, then the user may cannot check and manage it anymore. Maybe we can open an issue to trace it.

Comment (sfzeng) [reply file/pkg/service/service.go:129]: I created the issue: https://github.com/sodafoundation/multi-cloud/issues/1022

Comment (kumarashit) [reply file/pkg/service/service.go:129]: Agree. When we add more operations like Mounting and mount target, then we can delete from backednd too. Thanks @sfzeng for opening issue

**Pull request merged**